File: core/core-api/src/main/java/io/say/better/domain/symbol/application/impl/SymbolService.java
package io.say.better.domain.symbol.application.impl;

import java.util.List;

import org.springframework.stereotype.Service;

import io.say.better.storage.mysql.dao.repository.SymbolReadRepository;
import io.say.better.storage.mysql.dao.repository.SymbolWriteRepository;
import io.say.better.storage.mysql.domain.entity.Symbol;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Service
@RequiredArgsConstructor
public class SymbolService {

	private final SymbolReadRepository symbolReadRepository;
	private final SymbolWriteRepository symbolWriteRepository;

	public List<Symbol> getSymbols(List<String> symbols) {
		return symbolReadRepository.findByTitleIn(symbols);
	}

	public List<Symbol> getSymbols(String name) {
		return symbolReadRepository.findByTitleStartingWith(name);
	}
}

File: core/core-api/src/main/kotlin/io/say/better/domain/symbol/application/impl/SymbolService.kt
package io.say.better.domain.symbol.application.impl

import io.say.better.storage.mysql.domains.symbol.entity.Symbol
import io.say.better.storage.mysql.domains.symbol.repository.SymbolReadRepository
import io.say.better.storage.mysql.domains.symbol.repository.SymbolWriteRepository
import org.springframework.stereotype.Service

@Service
class SymbolService(
    private val symbolReadRepository: SymbolReadRepository,
    private val symbolWriteRepository: SymbolWriteRepository,
) {
    fun getSymbols(symbols: List<String>?): List<Symbol> {
        val symbolResults = hashSetOf<Symbol>()
        if (symbols == null) {
            return emptyList()
        }

        for (symbol in symbols) {
            val symbolResult = symbolReadRepository.findByTitleContaining(symbol)
            symbolResults.addAll(symbolResult)
        }
        return symbolResults.toList()
    }

    fun getSymbols(name: String): List<Symbol> = symbolReadRepository.findByTitleStartingWith(name)

    fun getSymbol(symbolId: Long): Symbol = symbolReadRepository.findById(symbolId).get()
}
